lib/lte_link_control/lte_lc.c

```diff
@@ -1520,14 +1536,14 @@ int lte_lc_periodic_search_set(const struct lte_lc_periodic_search_cfg *const cf
 
 	err = nrf_modem_at_printf(
 		"AT%%PERIODICSEARCHCONF=0," /* Write mode */
-		"%u,"  /* <loop> */
-		"%u,"  /* <return_to_pattern> */
-		"%u,"  /* <band_optimization> */
+		"%hu,"  /* <loop> */
+		"%hu,"  /* <return_to_pattern> */
+		"%hu,"  /* <band_optimization> */
 		"%s%s" /* <pattern_1> */
 		"%s%s" /* <pattern_2> */
 		"%s%s" /* <pattern_3> */
 		"%s",  /* <pattern_4> */
-		cfg->loop, cfg->return_to_pattern, cfg->band_optimization,
+		loop_tmp, cfg->return_to_pattern, cfg->band_optimization,
```

Comment:
Is temp variable usage necessary with `nrf_modem_at_printf()`?

Reply:
Yes. I tried passing in the value directly which caused `nrf_modem_at_printf` to fail.

Reply:
I think I got something mixed up. Need to do a few more tests.

Reply:
@tokangas Accidently passed in the pointer to `cfg->loop` when I initially tested this. Did a re-test and the API succeeds without using a temp variable. Updated the PR.